apps/testing/src/main/java/com/akto/rules/CreateAdminUserViaMassAssignment.java

```diff
@@ -0,0 +1,180 @@
+    }
+
+    protected Set<String> checkIfApiIsEligibleForTest(OriginalHttpRequest request, OriginalHttpResponse response) throws Exception{
+        if(response.getStatusCode() < 200 && response.getStatusCode() >= 300){
+            throw new Exception("Response code is not 200");
+        }
+        if(!allowedMethods.contains(request.getMethod())){
+            throw new Exception("Request type is not allowed");
+        }
+        if(StringUtils.isEmpty(request.getBody())){
+            throw new Exception("Request body is empty");
+        }
+        if(StringUtils.isEmpty(response.getBody())){
+            throw new Exception("Response body is empty");
+        }
+        boolean endpointNameMatched = stringMatchContains(request.getUrl(), allowedEndpointNames);
+        if(!endpointNameMatched){
+            throw new Exception("Doesn't match any of the allowed endpoint names");
+        }
+        boolean reqBodyContainsRequiredKeys = stringMatchContains(request.getBody(), allowedKeysInRequest);
+        if(!reqBodyContainsRequiredKeys){
+            throw new Exception("Request body doesn't contain any of the allowed keys");
+        }
+        boolean respBodyContainsRequiredKeys = stringMatchContains(response.getBody(), allowedKeysInResponse);
+        if(!respBodyContainsRequiredKeys){
+            throw new Exception("Response body doesn't contain any of the allowed keys");
+        }
+        Set<String> extraParams = checkForExtraParams(request, response);
+        if(extraParams.isEmpty()){
+            throw new Exception("No extra params found");
+        }
+        return extraParams;
+    }
+
+    private static boolean stringMatchContains(String str, List<String> allowedStrings) {
+        boolean stringMatched = false;
+        for(String endpointName: allowedStrings){
+            if (str.contains(endpointName)) {
+                stringMatched = true;
+                break;
+            }
+        }
+        return stringMatched;
+    }
+
+    private Set<String> checkForExtraParams(OriginalHttpRequest request, OriginalHttpResponse response) throws Exception{
+        BasicDBObject flattenedRequest =  JSONUtils.flattenWithDots(RequestTemplate.parseRequestPayload(request.getJsonRequestBody(), null));
+        BasicDBObject flattenedResponse = JSONUtils.flattenWithDots(RequestTemplate.parseRequestPayload(response.getJsonResponseBody(), null));
+        if(flattenedRequest.keySet().size() > flattenedResponse.keySet().size()){
+            throw new Exception("Request body has more params than response body");
+        }
+        Set<String> extraParams = new HashSet<>();
+        for(String responseKey: flattenedResponse.keySet()){
+            String[] splitResponseKeys = responseKey.split("\\.");
+            for(String splitResponseKey: splitResponseKeys){
+                for(String requestKey: flattenedRequest.keySet()){
+                    if(!requestKey.contains(splitResponseKey)){
```

Comment:
might not work properly in nested payloads, since here we are splitting the key. Only keys at root level will be detected. Do check once

Reply:
requestKey contains a dot seperated flattened key and splitResponseKey contains the actual leaf level key from response and since this a .contains check I dont think we will miss any keys.